src/store/module-map/actions.js

```diff
@@ -23,42 +23,61 @@ const addLayerActions = {
   tms: 'addLayerTMS',
   geojson: 'addLayerGeoJSON'
 }
-export function addLayer (context, { layerDescriptor, title }) {
+export function addLayer (context, { layerDescriptor, title, options }) {
   const type = layerDescriptor.type
   if (type && typeof type === 'string') {
     const action = addLayerActions[type.toLowerCase()]
     if (action) {
-      context.dispatch(action, { layerDescriptor, title })
+      context.dispatch(action, { layerDescriptor, title, options })
       return
     }
   }
 
   console.warn(`Trying to add layer of unknown type "${type}"`)
 }
 
-export function addLayerWMS (context, { layerDescriptor, title }) {
+function applyExtraOptions (defaultOptions, extraOptions, allowedOptions) {
+  const options = {}
+  if (typeof extraOptions !== 'undefined') {
+    for (const k in extraOptions) {
+      if (allowedOptions.indexOf(k) !== -1) {
+        options[k] = extraOptions[k]
+      }
+    }
+  }
+  return Object.assign(defaultOptions, options)
+}
+
+export function addLayerWMS (context, { layerDescriptor, title, options }) {
   const map = context.state.mapObject
-  const wms = L.tileLayer.wms(layerDescriptor.url, {
+  const defaultOptions = {
     layers: layerDescriptor.layers,
     format: 'image/png',
     transparent: true,
     maxZoom: 22
-  }).addTo(map)
+  }
+  const allowedOptions = ['minZoom', 'maxZoom', 'layers', 'styles', 'format', 'transparent', 'format', 'version',
+    'csr', 'upercase', 'attribution']
```

Comment:
uppercase

Reply:
some WMS services needs the variables in upercase